extensions/vscode/src/views/homeView.ts

```diff
@@ -1189,6 +1209,174 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
     }
   };
 
+  public async handleFileInitiatedDeployment(uri: Uri) {
+    // Guide the user to create a new Deployment with that file as the entrypoint
+    // if one doesn’t exist
+    // Select the Deployment with an active configuration for that entrypoint if there
+    // is only one
+    // With multiple, if a compatible one is already active, then do nothing.
+    // Otherwise, prompt for selection between multiple compatible deployments
+
+    // console.log("'Deploy with this Entrypoint' button hit!", uri);
+    const dir = relativeDir(uri);
+    // If the file is outside the workspace, it cannot be an entrypoint
+    if (dir === undefined) {
+      return false;
+    }
+    const entrypoint = uriUtils.basename(uri);
+
+    const api = await useApi();
+
+    // get all of the deployments for projectDir
+    // get the configs which are filtered for the entrypoint in that projectDir
+    // determine a list of deployments that use those filtered configs
+
+    let contentRecordList: (ContentRecord | PreContentRecord)[] = [];
+    const getContentRecords = new Promise<void>(async (resolve, reject) => {
+      try {
+        const response = await api.contentRecords.getAll({
+          dir,
```

Comment:
Noting this in case I'm misunderstanding - we don't use the entrypoint filtering here since it doesn't capture pre-deployment records atm?

Reply:
Correct. It is found in the configuration file associated with the content record only, within pre-deployment files. After deployment, the last entrypoint is recorded within the content record. This is the field that the backend API is using for filtering.